### PR TITLE
Improve backup performance with faster compression algorithm and multiple threads

### PR DESCRIPTION
**Pull request by derekbit**
https://github.com/longhorn/longhorn/issues/5189

Depends on https://github.com/longhorn/backupstore/pull/94

### CODE REVIEW

File: pkg/sync/backup.go [hunk 1]
```diff
@@ -53,27 +51,27 @@ func (t *Task) CreateBackup(backupName, snapshot, dest, backingImageName, backin
...
-	replicas, err := t.client.ReplicaList()
+	replica, err := t.findAvailableReplica()
 	if err != nil {
 		return nil, err
 	}
 
+	return t.createBackup(replica, backupName, snapshot, dest, volume.Name, backingImageName, backingImageChecksum, labels, credential)
+}
+
+func (t *Task) findAvailableReplica() (*types.ControllerReplicaInfo, error) {
```
Comment (shuo-wu) [inline pkg/sync/backup.go:64]: NIT: `findRWReplica()`

File: pkg/backup/main.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ var (
 	log     = logrus.WithFields(logrus.Fields{"pkg": "backup"})
 )
 
-type BackupParameters struct {
```
Comment (shuo-wu) [inline pkg/backup/main.go]: NIT: Retain the previous naming. This won't be used in the restore function

**Review by shuo-wu — DISMISSED** [review]
LGTM. Good refactoring! 
Don't miss the backward compatibility handling on the manager side.

File: pkg/backup/main.go (hunk 2 above)
Comment (derekbit) [reply pkg/backup/main.go]: Updated. Thank you.

**Review by shuo-wu — DISMISSED** [review]

**Comment by derekbit** [comment]
Dependencies are updated and ready for review.

File: pkg/backup/main.go [hunk 3]
```diff
@@ -21,6 +22,18 @@ var (
 	log     = logrus.WithFields(logrus.Fields{"pkg": "backup"})
 )
 
+type CreateBackupParameters struct {
```
Comment (innobead) [inline pkg/backup/main.go:25]: ref: https://stackoverflow.com/a/45886055 

I think the original naming should be fine. Or if you want it more precise, you can use `BackupCreateParameter`. 

nit: I feel we can do this pattern for the parameter for DoBackupRestore as well. This is not a priority. We can refactor over time.

**Review by innobead — APPROVED** [review]
LGTM.

File: pkg/backup/main.go (hunk 3 above)
Comment (derekbit) [reply pkg/backup/main.go:25]: `BackupCreateParameter` sounds good.

**Pull request merged**